File: tests/integration/construct_component/testcomponent/echo.ts
// Copyright 2016-2021, Pulumi Corporation.  All rights reserved.

import * as pulumi from "@pulumi/pulumi";

interface EchoArgs {
    echo: pulumi.Input<any>;
}

export class Echo extends pulumi.CustomResource {
    public readonly echo!: pulumi.Output<any>;
    constructor(name: string, args: EchoArgs, opts?: pulumi.CustomResourceOptions) {
        super("testprovider:index:Echo", name, args, opts);
    }
}

File: tests/integration/aliases/python/alias_after_failed_update/step1/echo.py
# Copyright 2016-2023, Pulumi Corporation.  All rights reserved.

from typing import Any, Optional

import pulumi

class Echo(pulumi.CustomResource):
    def __init__(self,
                 resource_name: str,
                 echo: pulumi.Input[Any],
                 opts: Optional[pulumi.ResourceOptions] = None):
        props = {
            "echo": echo,
        }
        super().__init__("testprovider:index:Echo", resource_name, props, opts)

    @property
    @pulumi.getter
    def echo(self) -> pulumi.Output[Any]:
        return pulumi.get(self, "echo")
